commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/SumTest.java

```diff
@@ -0,0 +1,181 @@
+            Arguments.of(new double[] {1, -7, 6}, 0),
+            Arguments.of(new double[] {1, 7, -15, 3}, -4),
+            Arguments.of(new double[] {2, 2, 2, 2}, 8.0),
+            Arguments.of(new double[] {2.3}, 2.3),
+            Arguments.of(new double[] {3.14, 2.718, 1.414}, 7.272),
+            Arguments.of(new double[] {12.5, 12.0, 11.8, 14.2, 14.9, 14.5, 21.0,
+                8.2, 10.3, 11.3, 14.1, 9.9, 12.2, 12.0, 12.1, 11.0, 19.8, 11.0, 10.0, 8.8,
+                9.0, 12.3}, 272.9),
+            Arguments.of(new double[] {-0.0, +0.0}, 0.0),
+            Arguments.of(new double[] {0.0, -0.0}, 0.0),
+            Arguments.of(new double[] {0.0, +0.0}, 0.0),
+            Arguments.of(new double[] {0.001, 0.0002, 0.00003, 10000.11, 0.000004}, 10000.111234),
+            Arguments.of(new double[] {10E-50, 5E-100, 25E-200, 35.345E-50}, 45.345E-50),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY},
+                Double.NaN),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.MAX_VALUE},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {Double.MAX_VALUE, Double.MAX_VALUE},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -Double.MAX_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, -Double.MIN_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.NaN, 34.56, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, Double.NaN, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, 89.74, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, 3.14, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.MAX_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.MAX_VALUE, 1}, Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, 1}, -Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -1, 1}, -Double.MAX_VALUE),
+            Arguments.of(new double[] {Double.MAX_VALUE, -1}, Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, -1}, -Double.MAX_VALUE),
+            Arguments.of(new double[] {1, 2, 3, Double.MAX_VALUE, -Double.MAX_VALUE, -7}, -1),
```

Comment:
Can you put this large cancellation test case in the array-of-arrays test as well, making sure the +/-MAX_VALUE is separated into the two arrays.